cwms-data-api/src/main/java/cwms/cda/api/BasinController.java

```diff
@@ -145,37 +178,219 @@ public void getOne(@NotNull Context ctx, @NotNull String basinId) {
             }
         } catch (SQLException ex) {
+            CdaError error = new CdaError("Error retrieving " + basinId);
             String errorMsg = "Error retrieving " + basinId;
             LOGGER.log(Level.SEVERE, errorMsg, ex);
+            ctx.status(HttpServletResponse.SC_INTERNAL_SERVER_ERROR).json(error);
         }
     }
 
-    @OpenApi(ignore = true)
+    @OpenApi(
+        queryParams = {
+            @OpenApiParam(name = NEW_BASIN_ID, required = true, description = "Specifies the"
+                    + " new name of the basin to be renamed."),
+            @OpenApiParam(name = OFFICE, required = true, description = "Specifies the"
+                    + " owning office of the basin to be renamed."),
+        },
+        responses = {
+            @OpenApiResponse(status = STATUS_404, description = "The provided combination of "
+                    + "parameters did not find a basin."),
+            @OpenApiResponse(status = STATUS_501, description = "Requested format is not "
+                    + "implemented")
+        },
+        description = "Renames CWMS Basin",
+        tags = {TAG}
+    )
     @Override
-    public void update(@NotNull Context ctx, @NotNull String s) {
-        ctx.status(HttpCode.NOT_IMPLEMENTED).json(CdaError.notImplemented());
+    public void update(@NotNull Context ctx, @NotNull String basinId) {
+        DSLContext dsl = getDslContext(ctx);
+        String formatHeader = ctx.header(Header.ACCEPT) != null ? ctx.header(Header.ACCEPT) :
+                Formats.NAMED_PGJSON;
+        ContentType contentType = Formats.parseHeader(formatHeader, Basin.class);
+        ctx.contentType(contentType.toString());
+
+        String officeId = ctx.queryParam(OFFICE);
+        String newBasinId = ctx.queryParam(NEW_BASIN_ID);
+
+        if (contentType.getType().equals(Formats.NAMED_PGJSON)) {
+            ctx.status(HttpCode.NOT_IMPLEMENTED).json(CdaError.notImplemented());
+        } else {
+            cwms.cda.data.dao.basin.BasinDao basinDao = new cwms.cda.data.dao.basin.BasinDao(dsl);
+            CwmsId oldLoc = new CwmsId.Builder()
+                    .withName(basinId)
+                    .withOfficeId(officeId)
+                    .build();
+            CwmsId newLoc = new CwmsId.Builder()
+                    .withName(newBasinId)
+                    .withOfficeId(officeId)
+                    .build();
+            try {
+                basinDao.renameBasin(oldLoc, newLoc);
+                ctx.status(HttpServletResponse.SC_OK).json("Updated Location");
+            } catch (IOException ex) {
+                CdaError error = new CdaError("Error renaming " + basinId);
+                String errorMsg = "Error renaming " + basinId;
+                LOGGER.log(Level.SEVERE, errorMsg, ex);
+                ctx.status(HttpServletResponse.SC_INTERNAL_SERVER_ERROR).json(error);
+            }
+
+        }
     }
 
-    @OpenApi(ignore = true)
+    @OpenApi(
+        queryParams = {
```

Comment:
These fields should be defined in the body, not as query params. Look at LookupTypeController or EmbankmentController to see how they do the create. You should use jackson to deserialize the json body content into a Basin DTO, and then pass that Basin DTO into the BasinDao's store method

Reply:
Addressed in latest commit